dashboard/src/components/__tests__/TimeDropdown.test.ts

```diff
@@ -1,114 +1,127 @@
 import { mount } from "@vue/test-utils";
+import VueDatePicker from "@vuepic/vue-datepicker";
 import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";
-import { ref } from "vue";
 
 import TimeDropdown from "../TimeDropdown.vue";
 
-const changeEvent = ref<{ field: string; value: string } | null>(null);
-
-const handleChange = (field: string, value: string) => {
-  changeEvent.value = { field, value };
-};
-
 describe("TimeDropdown.vue", () => {
+  let wrapper: ReturnType<typeof mount>;
+
   beforeEach(() => {
     vi.useFakeTimers();
+    wrapper = mount(TimeDropdown, {
+      attachTo: document.body,
+      props: {
+        name: "createdAt",
+      },
+    });
   });
 
   afterEach(() => {
     vi.useRealTimers();
-  });
-
-  it("renders correctly", () => {
-    const wrapper = mount(TimeDropdown, {
-      props: {
-        fieldname: "testField",
-        onChange: handleChange,
-      },
-    });
-    expect(wrapper.exists()).toBe(true);
+    wrapper.unmount();
   });
 
   it("initializes with correct default values", () => {
-    const wrapper = mount(TimeDropdown, {
-      props: {
-        fieldname: "testField",
-        onChange: handleChange,
-      },
-    });
     const button = wrapper.find("button");
+    const dropdown = wrapper.get("#time-filter-createdAt");
+
     expect(button.text()).toBe("Started");
+    expect(dropdown.isVisible()).toBe(false);
+  });
+
+  it("toggles dropdown visibility when button is clicked", async () => {
+    const dropdown = wrapper.get("#time-filter-createdAt");
+
+    await wrapper.get("#time-filter-createdAt-toggle").trigger("click");
+    expect(dropdown.isVisible()).toBe(true);
+
+    await wrapper.get("#time-filter-createdAt-toggle").trigger("click");
+    expect(dropdown.isVisible()).toBe(false);
+  });
```

Comment:
Should help testing the change to use the Dropdown component from Bootstrap, but not a lot of value after that.

Reply:
Removed it.